sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/PropertiesAccessCorrectFieldBase.cs

```diff
@@ -112,7 +112,7 @@ protected void CheckExpectedFieldIsUsed(IMethodSymbol methodSymbol, IFieldSymbol
             {
                 Location location = null;
                 string accessorType = null;
-                if (fields.Count() == 1)
+                if (fields.Count(x => x.UseFieldLocation) == 1)
```

Comment:
fields parameter is iterated here more than once. We should use a List instead of IEnumerable<>.